### PR TITLE
mgr/balancer: Python 3 compatibility fix

### PR DESCRIPTION
**Pull request by MariusSchiffer**
Fixes: http://tracker.ceph.com/issues/38831
Signed-off-by: Marius Schiffer <marius@mschiffer.de>

### CODE REVIEW

File: src/pybind/mgr/balancer/module.py [hunk 1]
```diff
@@ -928,7 +928,7 @@ def do_upmap(self, plan):
             if crush_rule not in pools_by_crush_rule:
                 pools_by_crush_rule[crush_rule] = []
             pools_by_crush_rule[crush_rule].append(pool)
-        classified_pools = pools_by_crush_rule.values()
+        classified_pools = list(pools_by_crush_rule.values())
```
Comment (tchaikov) [inline src/pybind/mgr/balancer/module.py]: yeah, we need to materialized the view before shuffling it.

**Comment by tchaikov** [comment]
@MariusSchiffer could you revise your commit message to put 
```
Fixes: http://tracker.ceph.com/issues/38831
```
right before your Signed-off-by line ?

**Comment by MariusSchiffer** [comment]
@tchaikov 
I have changed the commit message.
This should probably also be merged/backported into nautilus, as this was reintroduced by 3a730d751deff892e7a0bddba87eba3dbb829c3e

**Comment by tchaikov** [comment]
@MariusSchiffer thanks for the reminder. i just added "nautilus" to the tracker ticket's Backport field.

**Comment by sebastian-philipp** [comment]
mypy said:
```
pybind/mgr/balancer/module.py: note: In member "do_upmap" of class "Module":
pybind/mgr/balancer/module.py:933: error: Argument 1 to "shuffle" has incompatible type "ValuesView[Any]"; expected "List[Any]"
```

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-03-21_02:00:24-rados-wip-kefu-testing-2019-03-21-0010-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-03-22_03:22:17-rados-wip-kefu-testing-2019-03-21-0010-distro-basic-mira/

- the upgrade failures should be addressed by #27112
- the test_python.sh failures should be addressed by #27111

**Pull request merged**